vivarium/core/store.py

```diff
@@ -476,24 +476,45 @@ def _check_default(self, new_default):
                 str(self.default), str(new_default), str(new_default))
         return new_default
 
-    def _check_value(self, new_value):
+    def _check_schema(self, schema_key, new_schema):
         """Check a new schema value.
 
         Args:
-            new_value: The new value.
+            schema_key: the schema key
+            new_schema: The new schema value.
 
         Returns:
-            The new value.
+            The new schema value.
 
         Raises:
-            Exception: If the store already has a value and the new
+            Exception: If the store schema already has a value and the new
                 value is different from the existing one.
         """
-        if self.value is not None and new_value != self.value:
-            raise Exception(
-                '_value schema conflict: {} and {}'.format(
-                    new_value, self.value))
-        return new_value
+        current_schema = self.__dict__[schema_key]
+        if current_schema is not None and current_schema != new_schema:
+            raise ValueError(
+                f"incompatible {schema_key} schema assignment: {new_schema} "
+                f"at {self.path_for()}. "
+                f"schema {current_schema} is already assigned.")
+        return new_schema
+
+    def _check_schema_set_default(self, schema_key, new_schema, schema_registry):
+        current_schema = self.__dict__[schema_key]
+        if isinstance(new_schema, str):
+            new_schema = schema_registry.access(new_schema)
+        if isinstance(new_schema, dict) and isinstance(
+                new_schema[schema_key], str):
+            new_schema[schema_key] = schema_registry.access(
+                new_schema[schema_key])
+        if (
+                current_schema
+                and current_schema != DEFAULT_SCHEMA
+                and current_schema != new_schema):
+            raise ValueError(
+                f"incompatible {schema_key} schema assignment: {new_schema} "
+                f"at {self.path_for()}. "
+                f"schema {current_schema} is already assigned.")
```

Comment:
I think this error message could be clearer, e.g.:

`Incompatible schema assignment at {self.path_for()}. Trying to assign the value {new_schema} to key {schema_key}, which already has the value {current_schema}.`